src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,303 @@
+static constexpr uint8_t kMaxScenesPerFabric = CHIP_CONFIG_SCENES_MAX_PER_FABRIC;
+
+static constexpr size_t kIteratorsMax       = CHIP_CONFIG_MAX_SCENES_CONCURRENT_ITERATORS;
+static constexpr size_t kSceneNameMaxLength = CHIP_CONFIG_SCENES_CLUSTER_MAXIMUM_NAME_LENGTH;
+
+/// @brief SceneHandlers are meant as interface between various clusters and the Scene table.
+/// When a scene command involving extension field sets is received, the Scene Table will go through
+/// the list of handlers to either retrieve, populate or apply those extension field sets.
+///
+/// Generally, for each specific <endpoint, cluster> pair there should be one and only one handler
+/// registered with the scene table that claims to handle that pair.
+///
+/// A SceneHandler can handle a single <endpoint, cluster> pair, or many such pairs.
+///
+/// @note If more than one handler claims to handl a specific <endpoint, cluster> pair, only one of
+/// those handlers will get called when executing actions related to extension field sets on the scene
+/// table.  It is not defined which handler will be selected.
+
+class SceneHandler : public IntrusiveListNodeBase<>
+{
+public:
+    SceneHandler(){};
+    virtual ~SceneHandler() = default;
+
+    /// @brief Copies the list of supported clusters for an endpoint in a Span and resizes the span to fit the actual number of
+    /// supported clusters
+    /// @param endpoint target endpoint
+    /// @param clusterBuffer Buffer to hold the supported cluster IDs, cannot hold more than
+    /// CHIP_CONFIG_SCENES_MAX_CLUSTERS_PER_SCENE. The function shall use the reduce_size() method in the event it is supporting
+    /// less than CHIP_CONFIG_SCENES_MAX_CLUSTERS_PER_SCENE clusters.
+
+    virtual void GetSupportedClusters(EndpointId endpoint, Span<ClusterId> & clusterBuffer) = 0;
+
+    /// @brief Returns whether or not a cluster for scenes is supported on an endpoint
+    ///
+    /// @param endpoint Target Endpoint ID
+    /// @param cluster Target Cluster ID
+    /// @return true if supported, false if not supported
+    virtual bool SupportsCluster(EndpointId endpoint, ClusterId cluster) = 0;
+
+    /// @brief Called when handling AddScene.  Allows the handler to filter through the clusters in the command to serialize only
+    /// the supported ones.
+    ///
+    /// @param endpoint[in] Endpoint ID
+    /// @param extensionFieldSet[in] ExtensionFieldSets provided by the AddScene Command, pre initialized
+    /// @param cluster[out]  Cluster in the Extension field set, filled by the function
+    /// @param serialisedBytes[out] Buffer to fill from the ExtensionFieldSet in command
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR value otherwise
+    /// @note Only gets called after the scene-cluster has previously verified that the endpoint,cluster valuer pair is supported by
+    /// the handler. It is therefore the implementation's reponsibility to also implement the SupportsCluster method.

```

Comment:
I don't understand this part.  The cluster id is an outparam for this method.  If we already knew the cluster (so we could do the SupportsCluster() check), why do we need it as an outparam?